### PR TITLE
ISSUE-1788 Create SaaSDomainSubscriptionHandler and SaaSSubscriptionHandler

### PR DESCRIPTION
**Pull request by hungphan227**
refactor SaaSSubscriptionConsumer and SaaSDomainSubscriptionConsumer so that they can be reused in calendar service

https://github.com/linagora/twake-calendar-side-service/issues/565

## Summary by CodeRabbit

* **Refactor**
  * Consolidated subscription processing by moving domain and account logic into dedicated handler components and simplifying consumer dependencies.

* **New Features**
  * Added a dedicated subscription module that wires consumers, handlers, initialization steps, and health checks for improved operational visibility.

* **Tests**
  * Updated tests to construct consumers via the new handler-based approach and cover domain/account subscription flows.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                .doOnSuccess(success -> LOGGER.info("Cancelled SaaS subscription for domain: {}", domain));
+        }
+        return Mono.empty();
+    }
+
+    private Mono<Void> removeDomainIfExists(Domain domain) {
+        return Mono.fromRunnable(Throwing.runnable(() -> domainList.removeDomain(domain)))
+            .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
+            .then();
+    }
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImpl.java]: @hungphan227 please address this thanks!

File: tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImpl.java [hunk 2]
```diff
@@ -74,9 +74,15 @@ private Mono<Void> handleDomainCancelSubscriptionMessage(SaaSDomainSubscriptionM
...
-            .then();
+        return Mono.from(domainList.containsDomainReactive(domain))
+            .flatMap(alreadyExists -> {
+                if (!alreadyExists) {
+                    return Mono.empty();
+                }
+                return Mono.fromRunnable(Throwing.runnable(() -> domainList.removeDomain(domain)))
+                    .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
+                    .then();
+            });
```
[the remark this answers is not in the document]
Comment (hungphan227) [reply tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImpl.java]: SaaSDomainSubscriptionConsumerTest - verifyChainSaasDomainSubscriptionMessage

```
        String cancelMessage = String.format("""
            {
                "domain": "%s",
                "enabled": false
            }
            """, DOMAIN.asString());

        publishAmqpSaaSDomainSubscriptionMessage(cancelMessage);

        await.untilAsserted(() -> assertThat(domainList.containsDomain(DOMAIN)).isFalse());
```

[the remark this answers is not in the document]
Comment (Arsnael) [reply tmail-backend/saas/saas-rabbitmq/src/main/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImpl.java]: I think @chibenwa means a test where we cancel twice the same domain and make sure that the remove domain operation is idempotent and does not crash with an exception

File: tmail-backend/saas/saas-rabbitmq/src/test/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImplTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.linagora.tmail.rate.limiter.api.RateLimitingRepository;
+import com.linagora.tmail.rate.limiter.api.memory.MemoryRateLimitingRepository;
+import com.linagora.tmail.rate.limiter.api.model.RateLimitingDefinition;
+
+import reactor.core.publisher.Mono;
+
+public class SaaSDomainSubscriptionHandlerImplTest {
```
[the remark this answers is not in the document]
Comment (hungphan227) [reply tmail-backend/saas/saas-rabbitmq/src/test/java/com/linagora/tmail/saas/rabbitmq/subscription/SaaSDomainSubscriptionHandlerImplTest.java]: it is easier to write edge cases when testing SaaSDomainSubscriptionHandlerImpl directly
